src/web/app/pages-instructor/instructor-student-list-page/instructor-student-list-page.component.ts

```diff
@@ -67,461 +27,115 @@ interface GetCourseDetailsResponse {
-
   user: string = '';
-  isDisplayArchive: boolean = false;
-  courses: Course[] = [];
-
-  searchQuery: string = '';
-
-  courseDetailsList: CourseDetails[] = [];
-  sectionDetailsList: SectionDetails[] = [];
-  teamDetailsList: TeamDetails[] = [];
-  studentDetailsList: StudentDetails[] = [];
-
-  get allPresentCourses(): Course[] {
-    return this.courses.filter((course: Course) => !(course.isArchived && !this.isDisplayArchive));
-  }
-
-  get allCheckedCourses(): Course[] {
-    return this.allPresentCourses.filter((course: Course) => course.isChecked);
-  }
-
-  get allPresentCourseDetails(): CourseDetails[] {
-    const courseDetailsList: CourseDetails[] = [];
-
-    this.allPresentCourses.forEach((course: Course) =>  {
-      const courseDetails: CourseDetails | null = this.getCourseDetails(course.id);
-      if (courseDetails) {
-        courseDetailsList.push(courseDetails);
-      }
-    });
-
-    return courseDetailsList;
-  }
-
-  get allCheckedCourseDetails(): CourseDetails[] {
-    return this.allPresentCourseDetails.filter((courseDetails: CourseDetails) => courseDetails.isChecked);
-  }
-
-  get allPresentSectionDetails(): SectionDetails[] {
-    let allPresentSectionDetails: SectionDetails[] = [];
-
-    this.allCheckedCourseDetails.forEach((courseDetails: CourseDetails) => {
-      allPresentSectionDetails = allPresentSectionDetails.concat(courseDetails.sections);
-    });
-
-    return allPresentSectionDetails;
-  }
-
-  get allCheckedSectionDetails(): SectionDetails[] {
-    return this.allPresentSectionDetails.filter((sectionDetails: SectionDetails) => sectionDetails.isChecked);
-  }
-
-  get allPresentTeamDetails(): TeamDetails[] {
-    let allPresentTeamDetails: TeamDetails[] = [];
-
-    this.allCheckedSectionDetails.forEach((sectionDetails: SectionDetails) => {
-      allPresentTeamDetails = allPresentTeamDetails.concat(sectionDetails.teams);
-    });
-
-    return allPresentTeamDetails;
-  }
-
-  get allCheckedTeamDetails(): TeamDetails[] {
-    return this.allPresentTeamDetails.filter((teamDetails: TeamDetails) => teamDetails.isChecked);
-  }
-
-  get allPresentStudentDetails(): StudentDetails[] {
-    let allPresentStudentDetails: StudentDetails[] = [];
-
-    this.allCheckedTeamDetails.forEach((teamDetails: TeamDetails) => {
-      allPresentStudentDetails = allPresentStudentDetails.concat(teamDetails.students);
-    });
-
-    return allPresentStudentDetails;
-  }
-
-  get allCheckedStudentDetails(): StudentDetails[] {
-    return this.allPresentStudentDetails.filter((studentDetails: StudentDetails) => studentDetails.isChecked);
-  }
-
-  get isAllPresentCoursesChecked(): boolean {
-    return !!this.allPresentCourses.length && this.allPresentCourses.length === this.allCheckedCourses.length;
-  }
+  courseList!: Courses;
+  viewStudent!: boolean[];
+  courseInfo: {[key: string]: StudentListSectionData[]} = {};
+  courseStats: {[key: string]: Statistic} = {};
```

Comment:
While this is optional. I would recommend you to create a model interface instead of using multiple array and map to track the status.
For example, you can define an interface call `CourseTab`;
```java
interface CourseTab {
    studentListSectionDataList: StudentListSectionData[],
    hasTabExpanded: boolean;
    hasStudentLoaded: boolean; // to track whether the data has been loaded for performance optimization,
    stats: Statistic
}
```

You would use a variable called `courseTabList: CourseTab[]` in the class to track it.